src/api/api_client.rs

```diff
@@ -177,26 +177,32 @@ where
 	Runtime: FrameSystemConfig,
 {
 	/// Create a new Api client with call to the node to retrieve metadata.
-	pub fn new(client: Client) -> Result<Self> {
-		let genesis_hash = Self::get_genesis_hash(&client)?;
+	pub async fn new(client: Client) -> Result<Self> {
+		let genesis_hash_future = Self::get_genesis_hash(&client);
+		let metadata_future = Self::get_metadata(&client);
+		let runtime_version_future = Self::get_runtime_version(&client);
+
+		let (genesis_hash, metadata, runtime_version) = futures::future::try_join3(
```

Comment:
This unfortunately leads to problems with `no_std` 😢 
We could just make a couple of `await`s instead but it would be nice to wait in parallel.

Reply:
I think that's fine. The async improvement is worth the small duplication.